controller/share_manager_controller.go

```diff
@@ -300,9 +300,8 @@ func (c *ShareManagerController) syncShareManager(key string) (err error) {
 			return err
 		}
 
-		if err := c.ds.DeleteConfigMap(c.namespace, types.GetConfigMapNameFromShareManagerName(sm.Name)); err != nil {
-			return errors.Wrapf(err, "failed to delete the configmap (recovery backend) for share manager %v",
-				sm.Name)
+		if err := c.ds.DeleteConfigMap(c.namespace, types.GetConfigMapNameFromShareManagerName(sm.Name)); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
This kind of error seems appearing few times for different resource types previously. 

I would suggest to handle in datastore pkg directly, because this error doesn't mean a real error for us, or even I can say it's expected and intended. WDYT?

like: [/datastore/kubernetes.go#L138-L140](https://github.com/longhorn/longhorn-manager/blob/aa79220132ece38dbe55178104497748204e7d15/datastore/kubernetes.go#L138-L140)

Reply:
Updated. Thank you.